lib/Sequence.fz

```diff
@@ -902,6 +899,18 @@ public Sequence(public T type) ref is
   public type.AS_STRING_NON_FINITE_MAX_ELEMENTS => 10
 
 
+  # equality
+  #
+  public fixed redef type.equality(a, b Sequence T) bool
+    pre else
+      T : property.equatable
+      debug : a.finite
+      debug : b.finite
```

Comment:
This does not work like that since the inherited precondition is `true`, the resulting one is `true || T : property.equaltable && debug : a.finint && debug : b. fininte`, which is `true`. 
Apparently there is a bug the implementation of type constraints, the use of `x=y` below should cause an error since the precondition does not say anything about `T` being equatable.